### PR TITLE
Include knative info in metrics if present

### PR DESCRIPTION
**Pull request by sethvargo**
**Release Note**

```release-note
Include knative info in metrics if present
```

### CODE REVIEW

**Review by whaught — COMMENTED** [review]
/lgtm

File: pkg/observability/observability.go [hunk 1]
```diff
@@ -45,16 +55,31 @@ func WithRealmID(octx context.Context, realmID uint) context.Context {
...
+// to the observability context.
 func WithBuildInfo(octx context.Context) context.Context {
-	ctx, err := tag.New(octx,
-		tag.Upsert(BuildIDTagKey, buildinfo.BuildID),
-		tag.Upsert(BuildTagTagKey, buildinfo.BuildTag))
+	tags := make([]tag.Mutator, 0, 5)
+	tags = append(tags, tag.Upsert(BuildIDTagKey, buildinfo.BuildID))
+	tags = append(tags, tag.Upsert(BuildTagTagKey, buildinfo.BuildTag))
+
+	if knativeService != "" {
```
Comment (yegle) [inline pkg/observability/observability.go:65]: Either log it when it's empty, or add the tag even if the value is empty.

Per the Cloud Run contract, the env var should not be empty. When they are empty, we won't want to silently ignore the issue.

**Review by yegle — APPROVED** [review]

File: pkg/observability/observability.go (hunk 1 above)
Comment (sethvargo) [reply pkg/observability/observability.go:65]: We want to support cases where this runs outside of Cloud Run though. We know other people are running this code on other platforms. We aren't specifically tied to GCP.

**Pull request merged**